internal/controller/children.go

```diff
@@ -459,6 +461,61 @@ func (c ChildResourceUpdates) statusCallback(status *amaltheadevv1alpha1.Amalthe
 		c.StatefulSet.statusCallback(status)
 	}
 }
+func lastRequestTime(cr *amaltheadevv1alpha1.AmaltheaSession) (time.Time, error) {
+	url := fmt.Sprintf("http://%s:%s/request_stats", cr.Service().Name, string(amaltheadevv1alpha1.AuthProxyMetaPort))
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return time.Time{}, err
+	}
+	body, err := io.ReadAll(resp.Body)
+	defer func() {
+		err := resp.Body.Close()
+		if err != nil {
+			log.Log.Error(err, "couldn't close request body")
+		}
+	}()
+	if err != nil {
+		return time.Time{}, err
+	}
+	if resp.StatusCode != 200 {
+		return time.Time{}, fmt.Errorf("Couldn't get last request time: %s", body)
+	}
+	rt, err := time.Parse("2006-01-02 15:04:05", string(body))
+	if err != nil {
+		return time.Time{}, fmt.Errorf("Couldn't parse date from url %s : %w", url, err)
+	}
+	return rt, nil
+}
+
+func getIdleSince(
```

Comment:
Since the method returns two different, yet related, information (idle since and idle), I think `getIdleSince` is a bit misleading now.
Maybe something like:
```suggestion
func getIdleState(
```